lib/tls.js

```diff
@@ -204,7 +204,7 @@ exports.checkServerIdentity = function checkServerIdentity(hostname, cert) {
     if (!valid)
       reason = `IP: ${hostname} is not in the cert's list: ${ips.join(', ')}`;
     // TODO(bnoordhuis) Also check URI SANs that are IP addresses.
-  } else if (subject) {
+  } else if (subject || altNames) {
```

Comment:
Line 215 is going to throw a TypeError if it tries to read `subject.CN` when `subject == null` and `altNames` does not contain DNS:, IP: or URI: fields.

The logic should be something like `} else if (subject || haveAltNames) {` where `haveAltNames` is this:
```js
const haveAltNames = dnsNames.length > 0 || ips.length > 0 || uriNames.length > 0;
```
I'd swap the branches of the `if` statement on line 214 so you can write `if (haveAltNames) {` - negations are to be avoided when possible.

Reply:
I would have done the opposite: defined `const noAltNames` because switching the branches textually from their previous location makes the diff larger, and obscures what is actually being changed, which is much smaller.